### PR TITLE
stage2 ARM: Introduce MIR

### PR DESCRIPTION
**Pull request by joachimschmidt557**

### CODE REVIEW

File: src/arch/arm/Mir.zig [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        ///
+        /// Used by e.g. push
+        register_list: bits.Instruction.RegisterList,
+        /// Debug info: line and column
+        ///
+        /// Used by e.g. dbg_line
+        dbg_line_column: struct {
+            line: u32,
+            column: u32,
+        },
```
Comment (Luukdegram) [inline src/arch/arm/Mir.zig:182]: This makes `Data` 8 bytes instead of 4 as the doc comments above Data suggest.

Comment (andrewrk) [reply src/arch/arm/Mir.zig:182]: why not use the commented out comptime block below to assert that the size of Data is what it is expected to be?

Comment (joachimschmidt557) [reply src/arch/arm/Mir.zig:182]: Yes, `Data` is supposed to be 8 bytes. However, due to the use of tagged unions such as `bits.Instruction.Operand` etc. this is not the case yet and `Data` is currently 12 bytes large. Should I flatten down `Data` to 8 bytes here or can I do that in a follow-up PR?

Comment (andrewrk) [reply src/arch/arm/Mir.zig:182]: Fine by me to do in a follow-up PR, as long as you see it as a to-do item :)

Comment (joachimschmidt557) [reply src/arch/arm/Mir.zig:184]: Ok, I corrected the doc comment and added a TODO comment for fixing the size of `Data` in future.

**Pull request merged**